project-base/storefront/utils/useDeferredRender.ts

```diff
@@ -0,0 +1,116 @@
+import { NextRouter, useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { useSessionStore } from 'store/useSessionStore';
+import { FriendlyPagesDestinations } from 'types/friendlyUrl';
+
+type DeferPage = 'product' | 'category' | 'homepage';
+
+type DeferPlace = (
+    | typeof PRODUCT_PAGE_DEFER_ORDER
+    | typeof CATEGORY_PAGE_DEFER_ORDER
+    | typeof HOMEPAGE_DEFER_ORDER
+)[number];
+
+const getDeferPage = (router: NextRouter): DeferPage | 'non_deferred' => {
+    if (router.pathname === FriendlyPagesDestinations.product) {
+        return 'product';
+    } else if (
+        router.pathname === FriendlyPagesDestinations.category ||
+        router.pathname === FriendlyPagesDestinations.seo_category
+    ) {
+        return 'category';
+    } else if (router.pathname === '/') {
+        return 'homepage';
+    }
+
+    return 'non_deferred';
+};
+
+export const useDeferredRender = (place: DeferPlace) => {
+    const hadClientSideNavigation = useSessionStore((s) => s.hadClientSideNavigation);
+    const router = useRouter();
+    const page = getDeferPage(router);
+    const isDeferredPage = page !== 'non_deferred';
+    const [shouldRender, setShouldRender] = useState(!isDeferredPage || hadClientSideNavigation);
+
+    // eslint-disable-next-line consistent-return
+    useEffect(() => {
+        if (isDeferredPage) {
+            const defer = getDeferByPageAndPlace(page, place);
+            const timer = setTimeout(() => {
+                setShouldRender(true);
+            }, defer);
+
+            return () => {
+                clearTimeout(timer);
+            };
+        }
+    }, []);
```

Comment:
This way you can remove eslint comment

```suggestion
    useEffect(() => {
        let timer: NodeJS.Timeout | undefined;
        
        if (isDeferredPage) {
            const defer = getDeferByPageAndPlace(page, place);
            timer = setTimeout(() => {
                setShouldRender(true);
            }, defer);
        }

        return () => {
            clearTimeout(timer);
        };
    }, []);
```

Reply:
Thanks, changed.